src/fetch-cli.test.ts

```diff
@@ -0,0 +1,375 @@
+import { tmpdir } from 'node:os';
+import { join } from 'node:path';
+import { afterEach, beforeEach, describe, expect, it, jest } from 'bun:test';
+import { createFixture } from 'fs-fixture';
+import { loadDailyUsageData, loadSessionData } from './data-loader.ts';
+import { PricingFetcher } from './pricing-fetcher.ts';
+
+describe('--fetch CLI argument functionality', () => {
+	describe('PricingFetcher reuse prevents multiple fetch attempts', () => {
+		it('should demonstrate the fix prevents multiple URL fetch attempts', async () => {
+			// This test demonstrates that the issue is fixed by showing
+			// that PricingFetcher instances cache their results
+
+			const mockFetch = jest.fn()
+				.mockResolvedValueOnce({
+					ok: true,
+					json: async () => Promise.resolve({
+						'test-model': {
+							input_cost_per_token: 0.00001,
+							output_cost_per_token: 0.00003,
+						},
+					}),
+				});
+
+			const originalFetch = globalThis.fetch;
+			// eslint-disable-next-line ts/no-unsafe-assignment
+			globalThis.fetch = mockFetch as any;
+
+			try {
+				// Create a shared PricingFetcher instance
+				using pricingFetcher = new PricingFetcher('https://test-url.com/pricing.json');
+
+				// First call should trigger fetch
+				await pricingFetcher.fetchModelPricing();
+				expect(mockFetch).toHaveBeenCalledTimes(1);
+
+				// Second call should use cached result
+				await pricingFetcher.fetchModelPricing();
+				expect(mockFetch).toHaveBeenCalledTimes(1); // Still 1, not 2
+
+				// Third call should also use cached result
+				await pricingFetcher.getModelPricing('test-model');
+				expect(mockFetch).toHaveBeenCalledTimes(1); // Still 1, not 2
+			}
+			finally {
+				globalThis.fetch = originalFetch;
+			}
+		});
+
+		it('should verify that separate PricingFetcher instances fetch independently', async () => {
+			const mockFetch = jest.fn()
+				.mockResolvedValue({
+					ok: true,
+					json: async () => Promise.resolve({
+						'test-model': {
+							input_cost_per_token: 0.00001,
+							output_cost_per_token: 0.00003,
+						},
+					}),
+				});
+
+			const originalFetch = globalThis.fetch;
+			// eslint-disable-next-line ts/no-unsafe-assignment
+			globalThis.fetch = mockFetch as any;
+
+			try {
+				// First PricingFetcher instance
+				using fetcher1 = new PricingFetcher('https://test-url.com/pricing.json');
+				await fetcher1.fetchModelPricing();
+				expect(mockFetch).toHaveBeenCalledTimes(1);
+
+				// Second PricingFetcher instance (separate from first)
+				using fetcher2 = new PricingFetcher('https://test-url.com/pricing.json');
+				await fetcher2.fetchModelPricing();
+				expect(mockFetch).toHaveBeenCalledTimes(2); // Now 2 fetches
+			}
+			finally {
+				globalThis.fetch = originalFetch;
+			}
+		});
+	});
+
+	describe('Custom pricing source integration', () => {
+		let tempDir: string;
+		let testPricingFile: string;
+
+		beforeEach(async () => {
+			tempDir = tmpdir();
```

Comment:
don't use `tmpdir`. use `fs-fixture` and `await using` syntax as other test files do